src/Analyzer.Reports/ConsoleReportWriter.cs

```diff
@@ -17,61 +16,59 @@ public class ConsoleReportWriter : IReportWriter
         internal static string IndentedNewLine = Environment.NewLine + "\t";
         internal static string TwiceIndentedNewLine = Environment.NewLine + "\t\t";
 
+        private readonly List<string> filesAlreadyOutput = new List<string>();
+
         /// <inheritdoc/>
         public void WriteResults(IEnumerable<Types.IEvaluation> evaluations, IFileInfo templateFile, IFileInfo parametersFile = null)
         {
-            // Log info on file to be analyzed
-            string fileMetadata = Environment.NewLine + Environment.NewLine + $"File: {templateFile}";
-            if (parametersFile != null)
+            var resultsByFile = ReportsHelper.GetResultsByFile(evaluations, filesAlreadyOutput, out int passedEvaluations);
+
+            // output files in sorted order, but always output root first and regardless if no results
+            var filesWithResults = resultsByFile.Keys.ToList();
+            filesWithResults.Remove(templateFile.FullName);
+            filesWithResults.Sort();
+            filesWithResults.Insert(0, templateFile.FullName);
+
+            foreach(var fileWithResults in filesWithResults)
             {
-                fileMetadata += Environment.NewLine + $"Parameters File: {parametersFile}";
-            }
-            Console.WriteLine(fileMetadata);
+                string fileMetadata;
+                fileMetadata = Environment.NewLine + Environment.NewLine + $"File: {fileWithResults}";
 
-            OutputToConsole(evaluations);
-        }
+                if (fileWithResults == templateFile.FullName)
+                {
+                    if (parametersFile != null)
+                    {
+                        fileMetadata += Environment.NewLine + $"Parameters File: {parametersFile}";
```

Comment:
Oh, does this work okay after the changes on the PR to auto-discovery parameters files?

Reply:
I did some ad hoc testing and it seemed to work but we also need to think through how we want to show results for files that may be evaluated twice, e.g. main.bicep refers to module.bicep, but there's also main.parameters.json. So, when scanning thatdirectory, module.bicep is going to be scanned twice--once as a module referenced by main.bicep with the parameters from main.parameters.json, and once as a standalone bicep file with no parameter file for itself.

How do we output this and have it not be confusing?? Or do we dedupe and maybe only show discrepancies?

Reply:
mmm do we gain anything by scanning module.bicep by itself? Maybe we shouldn't even scan those files more than once?